core/model/modx/processors/resource/trash/purge.class.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * Empties the recycle bin.
+ *
+ * @return boolean
+ * @package    modx
+ * @subpackage processors.resource
+ */
+class modResourceTrashPurgeProcessor extends modProcessor {
+
+    /** @var array $resources */
+    public $resources;
+
+    /**
+     * @var array The ids of the resources to be deleted.
+     */
+    public $ids;
+
+    /** @var  array Failed ids of resources  */
+    private $failures;
+
+    public function checkPermissions() {
+        return $this->modx->hasPermission('purge_deleted');
+    }
+
+    public function getLanguageTopics() {
+        return array( 'resource','trash' );
+    }
+
+    /**
+     * @return bool|null|string
+     */
+    public function initialize() {
+        $idlist = $this->getProperty('ids', false);
+        //$this->modx->log(modX::LOG_LEVEL_DEBUG, "[purge] purging ids: ".$idlist);
+        if (!$idlist) {
+            return $this->modx->lexicon('resource_err_ns');
+        }
+        if ($idlist==-1) {
+            $this->modx->log(modX::LOG_LEVEL_WARN, "[purge] purging everything we are allowed to.");
```

Comment:
This (and a couple of other lines below) will send data to the MODX Error log (if WARN is enabled) - would be better to log to the manager log instead.